### PR TITLE
[FLINK-2013] Create generalized linear model framework

### PR DESCRIPTION
**Pull request by mtunique**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by mtunique** [comment]
Split from #3428.

**Comment by bowenli86** [comment]
@mtunique @thvasilo Hi guys, any update on this one?

File: flink-libraries/flink-ml/src/test/scala/org/apache/flink/ml/regression/MultipleLinearRegressionITSuite.scala [hunk 1]
```diff
@@ -75,15 +75,16 @@ class MultipleLinearRegressionITSuite
...
+    parameters.add(WithIterativeSolver.Stepsize, 2.0)
+    parameters.add(WithIterativeSolver.Iterations, 10)
+    parameters.add(WithIterativeSolver.ConvergenceThreshold, 0.001)
 
     mlr.fit(sparseInputDS, parameters)
 
     val weightList = mlr.weightsOption.get.collect()
 
     val WeightVector(weights, intercept) = weightList.head
+    println(weightList)
```
Comment (bowenli86) [inline flink-libraries/flink-ml/src/test/scala/org/apache/flink/ml/regression/MultipleLinearRegressionITSuite.scala]: remove this?

**Comment by bowenli86** [comment]
+1, LGTM

File: flink-libraries/flink-ml/src/test/scala/org/apache/flink/ml/regression/MultipleLinearRegressionITSuite.scala (hunk 1 above)
Comment (mtunique) [reply flink-libraries/flink-ml/src/test/scala/org/apache/flink/ml/regression/MultipleLinearRegressionITSuite.scala]: Ok I will delete  it.

**Comment by zentol** [comment]
Closing since flink-ml is effectively frozen.

**Pull request closed without merging**